chain/internal/rescan/rescan.go

```diff
@@ -18,7 +18,11 @@ type Interface interface {
 	Starter
 	Updater
 
+	// WaitForShutdown blocks until the underlying rescan object is shutdown.
```

Comment:
This commit should be squashed with the commit that introduced the code. A PR should be structured in a way that explains to the reviewer how the pieces fit together, not what versions of changes each file went through during development. So things like renames or adding comments to code that is introduced in the same PR should always be squashed with the commit that introduces the code. Otherwise a reviewer needs to keep all different names in memory, which makes it harder.

Reply:
noted, thanks